### ISSUE TITLE
[SubstrateVM] Cannot complete the Quick Start guide

### ISSUE THREAD

**Issue by Danstahr** [oracle/graal#560]
Versions
-----------------

**git log**
`commit bbb17b2a07784a77c037ac999330bc08ade073a5 (HEAD -> master, origin/master, origin/HEAD)`

**$JAVA_HOME/bin/java -version**
```
java version "1.8.0_172"
Java(TM) SE Runtime Environment (build 1.8.0_172-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.71-b01-internal-jvmci-0.46, mixed mode)
```
**mx --version**
`mx version 5.178.7`

Problem
-----------

Following the tutorial, after running `mx native-image`, the command fails with

```Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.NoClassDefFoundError: com/oracle/svm/core/option/HostedOptionKey
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.privateGetMethodRecursive(Class.java:3048)
	at java.lang.Class.getMethod0(Class.java:3018)
	at java.lang.Class.getMethod(Class.java:1784)
	at sun.launcher.LauncherHelper.validateMainClass(LauncherHelper.java:544)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:526)
Caused by: java.lang.ClassNotFoundException: com.oracle.svm.core.option.HostedOptionKey
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 7 more
```
`mx java HelloWorld` works fine.

**Comment by olpaw** [comment]
Could this be the same as https://github.com/oracle/graal/issues/559 ?

Please try if removing the svmbuild directory solves the problem.

**Comment by Danstahr** [comment]
Thanks Paul, removing the svmbuild directory does fix the problem.

Note: The linked issue says it's an OSX problem, I'm running Ubuntu though.

**Comment by olpaw** [comment]
> I will make mx native-image report a warning if the native-image-root folder was created with a different rev-id than the one currently using it.

Fixed in e28ecd600c3
